src/components/application_manager/src/commands/mobile/set_app_icon_request.cc

```diff
@@ -190,31 +190,41 @@ void SetAppIconRequest::CopyToIconStorage(
 void SetAppIconRequest::RemoveOldestIcons(const std::string& storage,
                                           const uint32_t icons_amount) const {
   const std::vector<std::string> icons_list = file_system::ListFiles(storage);
-  std::map<uint64_t, std::string> icon_modification_time;
+  std::vector<file_system::FileDescriptor> info_about_icons;
```

Comment:
@istoilovgithub Well the main reason I use std::sort for this std::vector is that I need to use custom compare function for this std::sort (to compare time stamps), is it possible to use such function with std::set?